Simple-Projects/Simple_Projects/Basic/List/List.cpp

```diff
@@ -72,28 +64,26 @@ void List::push_back(int value) {
        array = newArray;
     }
     array[current] = value;
-    current = newCurrent;
+	++current;
 }
 
 int List::pop_back() {
     if ( current == 0 ) {
        throw ZeroLenException();
     }
-    current -= 1;
-    return array[current];
+    return array[--current];
 }
 
 void List::sort() {
-    int last = current - 1;
-
-	for ( int i = 0; i < last; i++ ) {
-		for ( int j = 0; j < last; j++ ) {
-			int min = j + 1;
-            if ( array[j] > array[min] ) {
-                int temp = array[j];
-
-                array[j] = array[min];
-                array[min] = temp;
+    const int last = current - 1;
+
+	// Bubble sorting will keep an invariant as follows: [sorted|unsorted]
+	for ( int sorted = 0; sorted < last; ++sorted ) {
+		for ( int i = last; i > sorted; --i ) {
+			// Bubble down the max element from the unsorted region
+			const int prev = i - 1;
+            if ( array[prev] > array[i] ) {
+                std::swap(array[prev], array[i]);
```

Comment:
Using swap takes less text and is just as effective as making temporary variables yourself, because the code will most certainly inline.